src/tiled/mapdocument.cpp

```diff
@@ -125,6 +127,58 @@ MapDocument::~MapDocument()
     delete mMap;
 }
 
+/**
+ * Currently, this is the entry point for testing templates,
+ * it creates a templateGroup from the selected objects except the last object,
+ * then creates a templatGroupDocument from the templateGroup,
+ * then adds the last object to it, saves, loads then prints the object data
+ */
+bool MapDocument::saveSelectedObjectsAsTemplateGroup()
+{
+    TemplateGroupFormat *templateGroupFormat = new TtxTemplateGroupFormat();
+
+    TemplateGroup *templateGroup = new TemplateGroup(QLatin1String("test group"));
+    templateGroup->setFormat(templateGroupFormat);
+
+    for (MapObject *o : mSelectedObjects) {
+        auto tileset = o->cell().tileset();
+        if (tileset)
+            templateGroup->addTileset(tileset->sharedPointer());
+    }
+
+    for (int i = 0; i < mSelectedObjects.size() - 1; ++i) {
+        ObjectTemplate *objectTemplate = new ObjectTemplate(i+1, QLatin1String("template ") + QString::number(i+1));
+        objectTemplate->setObject(mSelectedObjects[i]);
+        templateGroup->addTemplate(objectTemplate);
+    }
+
+    QString fileName = QLatin1String("templateGroup.ttx");
+
+    TemplateGroupDocument *templateGroupDocument = new TemplateGroupDocument(templateGroup, fileName);
```

Comment:
No change required, but just to note, that to have less repetition I am often using `auto` for such lines these days, since the type is completely obvious:
```c++
auto templateGroupDocument = new TemplateGroupDocument(templateGroup, fileName);
```